store/postgres/src/mapping_sql.rs

```diff
@@ -0,0 +1,1107 @@
+/// at compile time. Because of that, we retrieve the actual data for an
+/// entity as Jsonb by converting the row containing the entity using the
+/// `to_jsonb` function.
+#[derive(QueryableByName)]
+pub struct EntityData {
+    #[sql_type = "Text"]
+    entity: String,
+    #[sql_type = "Jsonb"]
+    data: serde_json::Value,
+}
+
+impl EntityData {
+    fn value_from_json(
+        column_type: ColumnType,
+        json: serde_json::Value,
+    ) -> Result<graph::prelude::Value, StoreError> {
+        use graph::prelude::Value as g;
+        use serde_json::Value as j;
+        // Many possible conversion errors are already caught by how
+        // we define the schema; for example, we can only get a NULL for
+        // a column that is actually nullable
+        match json {
+            j::Null => Ok(g::Null),
+            j::Bool(b) => Ok(g::Bool(b)),
+            j::Number(number) => match column_type {
+                ColumnType::Int => match number.as_i64() {
+                    Some(i) => i32::try_from(i).map(|i| g::Int(i)).map_err(|e| {
+                        StoreError::Unknown(format_err!(
+                            "failed to convert {} to Int: {}",
+                            number,
+                            e
+                        ))
+                    }),
+                    None => Err(StoreError::Unknown(format_err!(
+                        "failed to convert {} to Int",
+                        number
+                    ))),
+                },
+                ColumnType::BigDecimal => {
+                    let s = number.to_string();
+                    scalar::BigDecimal::from_str(s.as_str())
+                        .map(|d| g::BigDecimal(d))
+                        .map_err(|e| {
+                            StoreError::Unknown(format_err!(
+                                "failed to convert {} to BigDecimal: {}",
+                                number,
+                                e
+                            ))
+                        })
+                }
+                ColumnType::BigInt => {
+                    let s = number.to_string();
+                    scalar::BigInt::from_str(s.as_str())
+                        .map(|d| g::BigInt(d))
+                        .map_err(|e| {
+                            StoreError::Unknown(format_err!(
+                                "failed to convert {} to BigInt: {}",
+                                number,
+                                e
+                            ))
+                        })
+                }
+                column_type => Err(StoreError::Unknown(format_err!(
+                    "can not convert number {} to {:?}",
+                    number,
+                    column_type
+                ))),
+            },
+            j::String(s) => match column_type {
+                ColumnType::String => Ok(g::String(s)),
+                ColumnType::Bytes => scalar::Bytes::from_str(s.trim_start_matches("\\x"))
```

Comment:
No, it's not. It's the [bytea hex format](https://www.postgresql.org/docs/9.6/datatype-binary.html) Things are set up so that `Bytes` get stored as `bytea`, i.e. an actual array of bytes, not the encoding of the bytes as a hex string. They get (de)serialized in the hex format.

Reply:
Ah thanks for clarifying.